web/modules/custom/convert_text/src/ConvertText.php

```diff
@@ -34,18 +37,92 @@ protected static function convert(string $source_text, string $field_type): stri
 
       case 'html':
         $converter = new CommonMarkConverter();
-        return $converter->convert($source_text)->getContent();
+        $content = $converter->convert($source_text)->getContent();
+        return self::addLinkItMarkup($content);
 
       default:
         throw new \Exception("Invalid \$field_type of $field_type given");
+    }
+  }
+
+  /**
+   * Update local link tags with linkit data attributes.
+   */
+  protected static function addLinkItMarkup(string $source_text): string {
+    $dom = new \DOMDocument();
+    $dom->loadHTML($source_text);
+
+    foreach ($dom->getElementsByTagName('a') as $link) {
+      $href = $link->getAttribute('href');
+      if (!$href) {
+        continue;
+      }
+
+      $host = parse_url($href, PHP_URL_HOST) ?? '';
+
+      if (in_array($host, ['', 'digital.gov', 'www.digital.gov'])) {
+        $alias = parse_url($href, PHP_URL_PATH);
 
+        $sysPath = \Drupal::service('path_alias.manager')->getPathByAlias($alias);
+
+        // If a link already has all the linkit attributes, leave it be.
+        if (
+          $link->hasAttribute('data-entity-type')
+          && $link->hasAttribute('data-entity-uuid')
+          && $link->hasAttribute('data-entity-substitution')
+        ) {
+          continue;
+        }
+
+        $url = Url::fromUserInput($sysPath);
+        switch ($url->getRouteName()) {
+          case 'entity.node.canonical':
+            $params = $url->getRouteParameters();
+            $node = Node::load($params['node']);
+            if (!$node) {
+              continue 2;
+            }
+            $entityType = 'node';
+            $uuid = $node->uuid();
+            break;
+
+          case 'entity.taxonomy_term.canonical':
+            $entityType = 'term';
+            $params = $url->getRouteParameters();
+            $term = Term::load($params['taxonomy_term']);
+            if (!$term) {
+              continue 2;
+            }
+            $uuid = $term->uuid();
+            break;
+
+          case '<front>':
+            // If someone links to the home page,
+            // we don't need to modify the link.
+            continue 2;
+
+          default:
+            // Do we want to log / warn here?
+            continue 2;
```

Comment:
I was trying to think of a reason they would get here, and I thought of redirects (/event) and 404s(/blah). Those both cause a error
`UnexpectedValueException: base:event has no corresponding route. in Drupal\Core\Url->getRouteName() (line 567 of core/lib/Drupal/Core/Url.php). `
You may want to all `isRouted()` before switching.
I would think 404s should just remain as is. 403s though, that should follow the chain to find it's source.

Reply:
Also found out it doesn't like mailto links as well `<a href="mailto:digitalgov@gsa.gov">Email</a>`

Reply:
both of these are fixed now